### ISSUE TITLE
Error on Direction values after DICOMOrient

### ISSUE THREAD

**Issue by plenost** [SimpleITK/SimpleITK#1800]
### Description

DICOMOrient and DICOMOrientFilter don't change the direction properly after processing

### Steps to Reproduce

1. Load an axial image with an orientation LPS (1,0,0,0,1,0,0,0,1)
2. Apply sitk.DICOMOrient(original_volume, "PIL")
3. Output image is correctly oriented but the output value of GetDirection() on the reoriented volume is incorrect (0,0,1,1,0,0,0,-1,0) instead of (0,1,0,0,0,-1,1,0,0)

```
import SimpleITK as sitk

dicom_dir = "MY_SERIES_FOLDER"

reader = sitk.ImageSeriesReader()
dicom_files = reader.GetGDCMSeriesFileNames(dicom_dir)
reader.SetFileNames(dicom_files)
original_volume = reader.Execute()
input_direction = original_volume.GetDirection()
input_orientation = sitk.DICOMOrientImageFilter_GetOrientationFromDirectionCosines(input_direction)


target_volume = sitk.DICOMOrient(original_volume, "PIL")

output_actual_direction = target_volume.GetDirection()
output_actual_orientation = sitk.DICOMOrientImageFilter_GetOrientationFromDirectionCosines(output_actual_direction)

output_expected_direction = [0,1,0,0,0,-1,1,0,0]
output_expected_orientation = sitk.DICOMOrientImageFilter_GetOrientationFromDirectionCosines(output_expected_direction)

print(f'Input Direction: {input_direction}')
print(f'Input Orientation: {input_orientation}')

print(f'Actual Output Direction: {output_actual_direction}')
print(f'Input Orientation: {output_actual_orientation}')

print(f'Expected Output Direction: {output_expected_direction}')
print(f'Input Orientation: {output_expected_orientation}')

```

Console Ouput:
```
Input Direction: (1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0)
Input Orientation: LPS
Actual Output Direction: (0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, -1.0, 0.0)
Actual Output Orientation: PIL
Expected Output Direction: [0, 1, 0, 0, 0, -1, 1, 0, 0]
Expected Output Orientation: SLA
```

### Expected behavior

From my understanding, DICOM orientation in LPS
Meaning L = [1,0,0], P=[0,1,0] and S=[0,0,1]
Thus, if I ask a PIL orientation, I expect the result to be: [P,I,L] = [0,1,0,0,0,-1,1,0,0]

### Actual behavior

The output is actually (0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, -1.0, 0.0) which, to me, is the equivalent of SLA
Yet the function DICOMOrientImageFilter_GetOrientationFromDirectionCosines gives me the opposite result.

### Reproducibility

100% reproducibility

### Versions

Tested with SimpleITK 2.2.0

### Environment

Using VS Code on Windows 10 with Python 3.10.7 64-bits

**Comment by plenost** [comment]
I am not sure I understand what you mean.

In the fundamental concepts as well as in the DICOM description, we have se following definition of the Image Orientation: "Direction cosine matrix (vector like type representing matrix in row major order) - direction of each of the axes corresponding to the matrix columns."

From what I understand, if the image orientation of the input series is [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]

- X component is represented by [1.0, 0.0, 0.0] (L)
- Y compenent is represented by [0.0, 1.0, 0.0] (P)
- Z component is represented by [0.0, 0.0, 1.0] (S)

Also this is the matrix representation of LPS. And so far I agree.

However, after re-orientation, I have:
- X component is represented by [0.0, 0.0, 1.0] (S)
- Y compenent is represented by [1.0, 0.0, 0.0] (L)
- Z component is represented by [0.0, -1.0, 0.0] (A)

I do not understand in which what the matrix above should be the equivalent to PIL. If we take the same approach than for the input matrix, I would obtain SLA.

**Comment by blowekamp** [comment]
Perhaps using this latex in Github help explain what I mean by column vectors, and you expectation of the matrix are transposed.

$$D =
\begin{bmatrix} 
  \begin{bmatrix} 
  0.0 \\
  1.0 \\
  0.0
  \end{bmatrix} 
  \begin{bmatrix} 
  0.0 \\
  0.0 \\
  -1.0
  \end{bmatrix} 
  \begin{bmatrix} 
  1.0 \\
  0.0 \\
  0.0
  \end{bmatrix} 
\end{bmatrix}$$

stored as array ordered as:

$$
\begin{bmatrix} 1 & 2 & 3 \\
		4 & 5 & 6 \\
                7 & 8 & 9
\end{bmatrix}
$$

This is constant with the results of the filter.

**Comment by plenost** [comment]
Welp, I'm dumb...
Thanks for the clarification and sorry for asking such a silly question.
I'll close the issue.

**Comment by b-niu** [comment]
Hi everyone! I've been following your discussion and noticed something peculiar about the DICOMOrient function. After extensive testing on my own, I found some strange results. Here's a summary of my findings:

| Original Orientation | Set to Orientation | Resulting Orientation |
|----------------------|--------------------|-----------------------|
| RAS                  | RAS                | LPS                   |
| LPS                  | LPS                | RAS                   |
| IAR                  | IAR                | IPL                   |

In my personal tests, it seems to always give the opposite result (except for S and I). Why is this happening?

![QQ_1721648168368](https://github.com/user-attachments/assets/f3246640-7758-4c3c-aada-083f1c6794d0)

**Comment by blowekamp** [comment]
This issue has been closed for 1.5 years. Your issue is different. Please create a new issue and provide the requested information for reproducibility.
